text/3366-diagnostic-attribute-namespace.md

```diff
@@ -71,32 +71,50 @@ I expect the new attributes to be documented on the existing [Diagnostics](https
+
 Any attribute in this namespace may:
 
 * Hint the compiler to emit a specific error message in a specific situation
+* Only affect the messages emitted by a compiler in case of a failed compilation
 
 Any attribute in this namespace is not allowed to:
 
 * Change the result of the compilation, which means applying such an attribute should never cause an error as long as they are syntactically valid
+* pass-through information from the source of the diagnostic in a way that users can rely on. E.g. Such an attribute should not allow users to keep the compilation successful and dump information about `extern` blocks to generate C header files
 
 The compiler is allowed to:
 
-* Ignore the provided hints
-* Use only parts of the provided hints, for example for the proposed `#[diagnostic::on_unimplemented]` only use the `message` option
-* Change this behaviour at any time
+* Ignore the hints provided by:
+    + A specific attribute
+    + A specific option 
+* Change the set of supported attributes and options at any time
+* Provide warning lints for malformed or otherwise not accepted attributes 
 
-The compiler must:
+The compiler must not:
 
-* Verify that the attribute is syntactically correct, which means:
-    + Its one of the accepted attributes
-    + It only contains the allowed options
-    + Any provided option follows the defined syntax
-* Follow the described semantics if the attribute is not ignored.
+* Change the semantic of an attribute or option, these kind of change requires a bump of the diagnostic attribute space version number
+* Emit an hard error on malformed attribute, although emitting lints is fine
 
+Adding a new attribute or option to the `#[diagnostic]` namespace is a decision of the compiler team. The envisioned procedure for this is as following:
+
+1. The new attribute lands on nightly without a feature gate. It gets ignored on stable/beta until its explicitly activated there.
+2. The time as nightly only attribute is used to verify that:
+    * The attribute solves a real problem
+    * The implemented syntax matches the expectations
+3. To enable a certain attribute in stable release the following steps are required:
+    * The attribute is documented in the rust reference
+    * T-lang is informed
+    * T-compiler performs a FCP, where a simple majority (>50%) accepts the attribute
```

Comment:
Leave a comment saying that the usual concern rules still apply, everyone can raise a concern.